packages/lambdas/src/fhir-to-medical-record.ts

```diff
@@ -0,0 +1,294 @@
+      });
+
+      if (entry) {
+        const value = entry.getElementsByTagName("value")[0];
+
+        const code = value.getAttribute("code");
+        const codeSystemName = value.getAttribute("codeSystemName");
+        const displayName = value.getAttribute("displayName");
+
+        const newCode = createCodeElement(code, codeSystemName, displayName, document);
+        codesRow.appendChild(newCode);
+
+        const translations = Array.from(value.getElementsByTagName("translation"));
+
+        translations.forEach(translation => {
+          const code = translation.getAttribute("code");
+          const codeSystemName = translation.getAttribute("codeSystemName");
+          const displayName = translation.getAttribute("displayName");
+
+          const newCode = createCodeElement(code, codeSystemName, displayName, document);
+          codesRow.appendChild(newCode);
+        });
+
+        tableCells.forEach(cell => {
+          if (cell.getAttribute("ID")?.includes("problemName")) {
+            const parentNode = cell.parentNode;
+            parentNode?.insertBefore(codesRow, cell.nextSibling);
+          }
+        });
+      }
+    }
+  }
+};
+
+const createCodeElement = (
+  code: string | null,
+  codeSystemName: string | null,
+  displayName: string | null,
+  document: Document
+) => {
+  const newCode = document.createElement("div");
+
+  const label = document.createElement("span");
+  label.setAttribute("class", "span_label");
+  label.appendChild(document.createTextNode(`${codeSystemName}: ${code}`));
+  newCode.appendChild(label);
+
+  newCode.appendChild(document.createElement("br"));
+
+  const display = document.createElement("span");
+  display.appendChild(document.createTextNode(`${displayName}`));
+  newCode.appendChild(display);
+
+  newCode.appendChild(document.createElement("br"));
+  newCode.appendChild(document.createElement("br"));
+
+  return newCode;
+};
+
+const createFileName = ({
+  cxId,
+  patientId,
+  resources,
+  dateFrom,
+  dateTo,
+}: {
+  cxId: string;
+  patientId: string;
+  resources?: ResourceTypeForConsolidation[];
+  dateFrom?: string;
+  dateTo?: string;
+}): string => {
+  const MEDICAL_RECORD_KEY = "MR";
+  let fileName = `${cxId}/${patientId}/${cxId}_${patientId}_${MEDICAL_RECORD_KEY}`;
+
+  if (resources) {
+    fileName = `${fileName}_${resources.toString()}`;
+  }
+
+  if (dateFrom) {
+    fileName = `${fileName}_${dateFrom}`;
+  }
+
+  if (dateTo) {
+    fileName = `${fileName}_${dateTo}`;
+  }
+
+  return `${fileName}.xml`;
+};
+
+const convertDoc = async ({
+  fileName,
+  conversionType,
+}: {
+  fileName: string;
+  conversionType?: string;
+}): Promise<string> => {
+  const result = await lambdaClient
+    .invoke({
+      FunctionName: convertDocLambda,
+      InvocationType: "RequestResponse",
+      Payload: JSON.stringify({ fileName, conversionType }),
+    })
+    .promise();
+
+  if (result.StatusCode !== 200) throw new Error("Lambda invocation failed");
```

Comment:
nit: add a bit more context on the error, this could lead the person looking at the error to think the whole lambda failed, not the one being called from this. Something like `"Error from conversion lambda"`. If its specific to PDF conversion, could add it there too.